File: C/exo5_1.c
#include <stdio.h>
#include <stdlib.h>
#include <pthread.h>

typedef struct
{
    int *tab;
    int taille;
}tableau;

void *afficheTableau(void *arg)
{
    tableau *a=(tableau *)arg;
    for(int i=0; i<a->taille; i++)
        printf("%d  ", a->tab[i]);
    printf("\n");
    return NULL;
}

int main()
{
    pthread_t th;
    tableau A;
    int temp[]={21, 20, 36, 5, 11};

    A.tab=temp;
    A.taille=5;
    pthread_create(&th, NULL, afficheTableau, &A);
    pthread_join(th, NULL);
    return 0;
}

File: python/exo5_1.py
import threading

class tableau :
    def __init__(self, tab, taille) :
        self.tab = tab
        self.taille = taille

def afficheTableau(arg) :
        a = arg
        for i in range(a.taille) :
            print(a.tab[i], end="  ")
        print()

A = tableau([21, 20, 36, 5, 11], 5)

th = threading.Thread(target=afficheTableau, args=(A,))
th.start()
th.join()